src/ThreeDSMethod.res

```diff
@@ -0,0 +1,111 @@
+open Utils
+@react.component
+let make = () => {
+  let (expiryTime, setExpiryTime) = React.useState(_ => 900000.0)
+  let (openModal, setOpenModal) = React.useState(_ => false)
+  let logger = Recoil.useRecoilValueFromAtom(RecoilAtoms.loggerAtom)
+
+  let mountToInnerHTML = innerHTML => {
+    let ele = Window.querySelector("#threeDsInvisibleIframe")
+    switch ele->Js.Nullable.toOption {
+    | Some(elem) => elem->Window.innerHTML(innerHTML)
+    | None =>
+      Js.Console.warn(
+        "INTEGRATION ERROR: Div does not seem to exist on which threeDSMethod is to be mounted",
+      )
+    }
+  }
+
+  React.useEffect0(() => {
+    handlePostMessage([("iframeMountedCallback", true->Js.Json.boolean)])
+    let handle = (ev: Window.event) => {
+      let json = ev.data->Js.Json.parseExn
+      let dict = json->Utils.getDictFromJson
+      if dict->Js.Dict.get("fullScreenIframeMounted")->Belt.Option.isSome {
+        let metadata = dict->getJsonObjectFromDict("metadata")
+        let metaDataDict =
+          metadata->Js.Json.decodeObject->Belt.Option.getWithDefault(Js.Dict.empty())
+        let threeDsDataDict =
+          metaDataDict
+          ->Js.Dict.get("threeDSData")
+          ->Belt.Option.flatMap(Js.Json.decodeObject)
+          ->Belt.Option.getWithDefault(Js.Dict.empty())
+        let threeDsUrl =
+          threeDsDataDict
+          ->Js.Dict.get("three_ds_method_details")
+          ->Belt.Option.flatMap(Js.Json.decodeObject)
+          ->Belt.Option.flatMap(x => x->Js.Dict.get("three_ds_method_url"))
+          ->Belt.Option.flatMap(Js.Json.decodeString)
+          ->Belt.Option.getWithDefault("")
+        let threeDsMethodData =
+          threeDsDataDict
+          ->Js.Dict.get("three_ds_method_details")
+          ->Belt.Option.flatMap(Js.Json.decodeObject)
+          ->Belt.Option.flatMap(x => x->Js.Dict.get("three_ds_method_data"))
+          ->Belt.Option.getWithDefault(Js.Dict.empty()->Js.Json.object_)
```

Comment:
You can get rid of the use of `Belt.` over here, since we are now have rescript-core integration